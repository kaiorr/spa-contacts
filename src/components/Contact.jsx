import React from 'react';

import './Contact.scss';

function ContactImage(props) {
  return (
    <span className="contact__avatar">
      <img src={props.image} alt={props.name}></img>
    </span>
  );
}

function ContactText(props) {
  return (
  <span className='contact__data'>{props.value}</span>
  );
}

export default function Contact(props) {
  return (
      <article className="contact" data-testid="contact">
        <ContactImage image={props.data.avatar} name={props.data.name} />
        <ContactText value={props.data.name} />
        <ContactText value={props.data.phone} />
        <ContactText value={props.data.country} />
        <ContactText value={props.data.admissionDate} />
        <ContactText value={props.data.company} />
        <ContactText value={props.data.department} />
      </article>
  );
}