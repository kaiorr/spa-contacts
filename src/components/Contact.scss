.contact {
  align-items: center;
  background-color: white;
  border-radius: 5px;
  border: 1px solid rgba(0, 0, 0, .1);
  box-shadow: 0 2px 2px rgba(0, 0, 0, .07);
  display: flex;
  justify-content: space-between;
  flex-wrap: nowrap;
  font-size: 1.2rem;
  margin-bottom: 20px;
  overflow: hidden;
  padding: 5px;

  &:first-of-type {
    margin-bottom: 10px;
    font-size: 20px;
    background-color: transparent;
    border: none;
    box-shadow: none;
    padding: 0 5px;
    font-weight: bold;

    & .contact__avatar {
      height: 0;
      border: none;
    }

    .contact__data:nth-child(2) {
      color: #000;
    }
  }
}

.contact__data {
  width: calc(100% / 6);
  padding: 0 10px;
}

.contact__data:nth-child(2) {
  color: #25366f;
}