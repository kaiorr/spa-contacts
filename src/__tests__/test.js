import React from 'react';
import {render, fireEvent, screen} from '@testing-library/react';

import Contact from '../components/Contact';

const mockUserData = {
	avatar: '',
	name: '',
	phone: '',
	country: '',
	admissionDate: '',
	company: '',
	department: '',
};

describe('Contacts', () => {
	it('Should render Contact', () => {
		const { fragment } = render(<Contact data={mockUserData}/>);

		expect(fragment).toMatchSnapshot();
    });
}
);