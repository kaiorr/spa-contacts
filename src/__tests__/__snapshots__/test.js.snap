// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contacts Should render Contact 1`] = `undefined`;
